userspace/engine/formats.cpp

```diff
@@ -94,43 +84,54 @@ std::string falco_formats::format_event(sinsp_evt *evt, const std::string &rule,
 
 		if(m_json_include_tags_property)
 		{
-			if (tags.size() == 0)
+			event["tags"] = tags;
+		}
+
+		event["output_fields"] = nlohmann::json::parse(json_fields);
+
+		for (auto const& ef : extra_fields)
+		{
+			std::string fformat = ef.second.first;
+			if (fformat.size() == 0)
 			{
-				// This sets an empty array
-				rule_tags = Json::arrayValue;
+				continue;
 			}
-			else
+
+			if (!(fformat[0] == '*'))
 			{
-				for (const auto &tag : tags)
-				{
-					rule_tags[rule_tags_idx++] = tag;
-				}
+				fformat = "*" + fformat;
 			}
-			event["tags"] = rule_tags;
-		}
 
-		full_line = writer.write(event);
-
-		// Json::FastWriter may add a trailing newline. If it
-		// does, remove it.
-		if(full_line[full_line.length() - 1] == '\n')
-		{
-			full_line.resize(full_line.length() - 1);
+			if(ef.second.second) // raw field
+			{
+				std::string json_field_map;
+				formatter = m_falco_engine->create_formatter(source, fformat);
```

Comment:
I am wondering whether `create_formatter` is a slow process; perhaps we might cache formatters per `<source,fformat>` key pair?